go/common/types.go

```diff
@@ -185,15 +185,15 @@ func (h *Header) Hash() L2RootHash {
 }
 
 type LogSubscription struct {
-	Accounts []*SubscriptionAccount
-	Filters  []*filters.FilterCriteria
+	SubscriptionAccount *SubscriptionAccount
+	Filter              *filters.FilterCriteria
 }
 
 // SubscriptionAccount is an authenticated account used when subscribing to logs.
 type SubscriptionAccount struct {
 	// The account the events relate to.
-	Account common.Address
-	// A signature over the subscription ID using the private viewing key. Prevents attackers from subscribing to
+	Account *common.Address
+	// A signature over the account address using the private viewing key. Prevents attackers from subscribing to
```

Comment:
In practice, it was easier to sign the account address than the subscription ID (since otherwise, we needed to generate the subscription ID back on the client, so it could be signed, which was feasible but a pain). I think the security properties are the same.

Reply:
umm. I proposed the subscription ID to prevent replay attacks. 
But I don't think they'd prevent it either, as someone could reuse that request on a different node.
Let's leave it like this for now. Pls add a todo with this concern

Reply:
Yeah, exactly. Sure, I'll add a todo.